source/lib/notifications-service.js

```diff
@@ -103,15 +103,27 @@ export function getNotificationObject(notificationInfo) {
 	};
 }
 
-export async function showNotifications(notifications) {
-	for (const notification of notifications) {
-		const notificationId = `github-notifier-${notification.id}`;
-		const notificationObject = getNotificationObject(notification);
+async function createNotification(notification) {
+	const notificationId = `github-notifier-${notification.id}`;
+	const notificationObject = getNotificationObject(notification);
 
-		await browser.notifications.create(notificationId, notificationObject);
-		await localStore.set(notificationId, notification);
+	await browser.notifications.create(notificationId, notificationObject);
+	await localStore.set(notificationId, notification);
+}
 
-		await delay(50);
+export async function showNotifications(notifications) {
+	for (const notification of notifications) {
+		try {
+			const urlToOpen = await notificationHandlers[notification.subject.type](notification);
+			const tabs = await queryTabs([urlToOpen]);
```

Comment:
`queryTabs` fails if the user has not granted the `tabs` permission, which we currently request for the "Reuse tabs" option.

If we assume that this permission might not be granted, then additional checks need to be applied here and also end up making this an option to request that permission (take a look at https://github.com/sindresorhus/notifier-for-github/pull/161, https://github.com/sindresorhus/notifier-for-github/pull/162, and https://github.com/sindresorhus/notifier-for-github/commit/a7af74c2adada9b73fc786649a903fbef548bd01 for context on asking permissions).

Both of the options above, I'm opposed to adding right now. And `api.openTab` already switches to a tab if we have one already open.

Thanks for the PR, but showing a notification with the existing behavior seems logical than not showing one. I'm gonna pass out on this one.